### PR TITLE
LibGUI: Stop the automatic autocomplete timer when typing whitespace / Playground: Improve the autocompleted suggestions

### PR DESCRIPTION
**Pull request by alimpfard**
Hopefully now less in-your-way than before!

---

There's no reason to bring up autocomplete when the user is just
inserting indentation and spaces and such.

- Now sorted
- Add a "layout" property to GUI::Widget and GUI::Frame
- Only complete Layouts for the values of "layout"
- Don't suggest anything if the only suggestion is what the user has
  already typed

### CODE REVIEW

File: AK/StringUtils.cpp [hunk 1]
```diff
@@ -36,6 +36,16 @@ namespace AK {
...
 namespace StringUtils {
 
+static constexpr bool is_whitespace_character(char ch)
+{
+    return ch == '\t'
+        || ch == '\n'
+        || ch == '\v'
+        || ch == '\f'
+        || ch == '\r'
+        || ch == ' ';
```
Comment (ADKaster) [inline AK/StringUtils.cpp:46]: `#include <ctype.h>` ? this looks like `isspace`

Comment (alimpfard) [reply AK/StringUtils.cpp:46]: Not sure if I _can_ do that in that header, maybe there's a reason why it was like that?
I'll try

Comment (alimpfard) [reply AK/StringUtils.cpp:46]: Build seems okay with it, I'll just use isspace :shrug:

**Comment by awesomekling** [comment]
LGTM but let's move the layout registrations to BoxLayout.cpp per Andrew's change :)

**Pull request merged**